From ac0d208b54fe78ff378afce8003467c763c6a74a Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Fri, 18 Mar 2022 10:10:53 -0300
Subject: [PATCH] misc: Use 64 bit time_t interfaces on syslog

It also handles the highly unlikely case where localtime might return
NULL, in this case only the PRI is set to hopefully instruct the relay
to get eh TIMESTAMP (as defined by the RFC).

Checked on x86_64-linux-gnu and i686-linux-gnu.
---
misc/syslog.c | 35 +++++++++++++++++++++++++++--------
1 file changed, 27 insertions(+), 8 deletions(-)

diff --git a/misc/syslog.c b/misc/syslog.c
index ee83b1bb76..fca3f6a6fa 100644
--- a/misc/syslog.c
+++ b/misc/syslog.c
@@ -140,7 +140,7 @@ __vsyslog_internal(int pri, const char *fmt, va_list ap,
 		   unsigned int mode_flags)
 {
 	struct tm now_tm;
-	time_t now;
+	__time64_t now;
 	int fd;
 	FILE *f;
 	char *buf = 0;
@@ -201,12 +201,12 @@ __vsyslog_internal(int pri, const char *fmt, va_list ap,
 	  {
 	    __fsetlocking (f, FSETLOCKING_BYCALLER);
 	    fprintf (f, "<%d>", pri);
-	    now = time_now ();
+	    now = time64_now ();
 	    f->_IO_write_ptr += __strftime_l (f->_IO_write_ptr,
 					      f->_IO_write_end
 					      - f->_IO_write_ptr,
 					      "%h %e %T ",
-					      __localtime_r (&now, &now_tm),
+					      __localtime64_r (&now, &now_tm),
 					      _nl_C_locobj_ptr);
 	    msgoff = ftell (f);
 	    if (LogTag == NULL)
